azkaban-common/src/main/java/azkaban/utils/StringUtils.java

```diff
@@ -88,4 +89,19 @@ public static String join2(Collection<String> list, String delimiter) {
 
     return buffer.toString();
   }
+
+  private static final Pattern BROWSWER_PATTERN = Pattern
+      .compile(".*Chrome.*|.*Firefox.*|.*Safari.*");
```

Comment:
Its better to use browser engine instead of browser name https://developer.mozilla.org/en-US/docs/Browser_detection_using_the_user_agent